lms/djangoapps/coursewarehistoryextended/fields.py

```diff
@@ -0,0 +1,21 @@
+"""
+Custom fields for use in the coursewarehistoryextended django app.
+"""
+
+from django.db.models.fields import AutoField
+
+
+class UnsignedBigIntAutoField(AutoField):
+    """
+    An unsigned 8-byte integer for auto-incrementing primary keys.
+    """
+    def db_type(self, connection):
+        if connection.settings_dict['ENGINE'] == 'django.db.backends.mysql':
+            return "bigint UNSIGNED AUTO_INCREMENT"
+        elif connection.settings_dict['ENGINE'] == 'django.db.backends.sqlite3':
+            # Sqlite will only auto-increment the ROWID column. Any INTEGER PRIMARY KEY column
+            # is an alias for that (https://www.sqlite.org/autoinc.html). An unsigned integer
+            # isn't an alias for ROWID, so we have to give up on the unsigned part.
+            return "integer"
+        else:
+            return None
```

Comment:
Can we make the fallthrough case something that won't choke on PostgreSQL? I know it's not a common use case, but when I tried it half a year ago, it did actually work, and I'd hate to lose that.